File: src/main/java/org/demo/InitialData.java
package org.demo;

import lombok.extern.slf4j.Slf4j;
import org.axonframework.commandhandling.gateway.CommandGateway;
import org.axonframework.common.IdentifierFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.event.ApplicationPreparedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
@Slf4j
public class InitialData {
	private final CommandGateway commandGateway;
	private final IdentifierFactory identifierFactory;

	@Autowired
	public InitialData(CommandGateway commandGateway, IdentifierFactory identifierFactory) {
		this.commandGateway = commandGateway;
		this.identifierFactory = identifierFactory;
	}

	@EventListener(ApplicationPreparedEvent.class)
	public void initialize() {
		log.info("Application prepared, adding initial data");
	}
}

File: src/main/java/org/demo/InitialData.kt
package org.demo

import mu.KotlinLogging
import org.axonframework.commandhandling.gateway.CommandGateway
import org.axonframework.common.IdentifierFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.event.ApplicationPreparedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class InitialData @Autowired constructor(private val commandGateway: CommandGateway, private val identifierFactory: IdentifierFactory) {
	private val log = KotlinLogging.logger {}

    @EventListener(ApplicationPreparedEvent::class)
    fun initialize() {
        log.info("Application prepared, adding initial data")
    }
}
